liveobjects/src/main/kotlin/io/ably/lib/liveobjects/DefaultRealtimeObject.kt

```
@@ -20,22 +43,323 @@ internal class DefaultRealtimeObject(
+            // RTO4b5 removed — buffer already cleared by RTO4d above
+            // defer the state change event until the next tick if we started a new sequence just now due to being in initialized state.
+            // this allows any event listeners to process the start of the new sequence event that was emitted earlier during this event loop.
+            objectsManager.endSync() // RTO4b4
+          }
+        }
+        ChannelState.detached,
+        ChannelState.suspended,
+        ChannelState.failed -> {
+          val errorReason = try {
+            adapter.getChannel(channelName).reason
+          } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)